File: distribution/python/dqops/client/models/default_column_checks_pattern_model.py
from typing import TYPE_CHECKING, Any, Dict, List, Type, TypeVar, Union

from attrs import define as _attrs_define
from attrs import field as _attrs_field

from ..types import UNSET, Unset

if TYPE_CHECKING:
    from ..models.column_default_checks_pattern_spec import (
        ColumnDefaultChecksPatternSpec,
    )


T = TypeVar("T", bound="DefaultColumnChecksPatternModel")


@_attrs_define
class DefaultColumnChecksPatternModel:
    """Default column-level checks pattern model

    Attributes:
        pattern_name (Union[Unset, str]): Pattern name
        pattern_spec (Union[Unset, ColumnDefaultChecksPatternSpec]):
        can_edit (Union[Unset, bool]): Boolean flag that decides if the current user can update or delete this object.
        yaml_parsing_error (Union[Unset, str]): Optional parsing error that was captured when parsing the YAML file.
            This field is null when the YAML file is valid. If an error was captured, this field returns the file parsing
            error message and the file location.
    """

    pattern_name: Union[Unset, str] = UNSET
    pattern_spec: Union[Unset, "ColumnDefaultChecksPatternSpec"] = UNSET
    can_edit: Union[Unset, bool] = UNSET
    yaml_parsing_error: Union[Unset, str] = UNSET
    additional_properties: Dict[str, Any] = _attrs_field(init=False, factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        pattern_name = self.pattern_name
        pattern_spec: Union[Unset, Dict[str, Any]] = UNSET
        if not isinstance(self.pattern_spec, Unset):
            pattern_spec = self.pattern_spec.to_dict()

        can_edit = self.can_edit
        yaml_parsing_error = self.yaml_parsing_error

        field_dict: Dict[str, Any] = {}
        field_dict.update(self.additional_properties)
        field_dict.update({})
        if pattern_name is not UNSET:
            field_dict["pattern_name"] = pattern_name
        if pattern_spec is not UNSET:
            field_dict["pattern_spec"] = pattern_spec
        if can_edit is not UNSET:
            field_dict["can_edit"] = can_edit
        if yaml_parsing_error is not UNSET:
            field_dict["yaml_parsing_error"] = yaml_parsing_error

        return field_dict

    @classmethod
    def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
        from ..models.column_default_checks_pattern_spec import (
            ColumnDefaultChecksPatternSpec,
        )

        d = src_dict.copy()
        pattern_name = d.pop("pattern_name", UNSET)

        _pattern_spec = d.pop("pattern_spec", UNSET)
        pattern_spec: Union[Unset, ColumnDefaultChecksPatternSpec]
        if isinstance(_pattern_spec, Unset):
            pattern_spec = UNSET
        else:
            pattern_spec = ColumnDefaultChecksPatternSpec.from_dict(_pattern_spec)

        can_edit = d.pop("can_edit", UNSET)

        yaml_parsing_error = d.pop("yaml_parsing_error", UNSET)

        default_column_checks_pattern_model = cls(
            pattern_name=pattern_name,
            pattern_spec=pattern_spec,
            can_edit=can_edit,
            yaml_parsing_error=yaml_parsing_error,
        )

        default_column_checks_pattern_model.additional_properties = d
        return default_column_checks_pattern_model

    @property
    def additional_keys(self) -> List[str]:
        return list(self.additional_properties.keys())

    def __getitem__(self, key: str) -> Any:
        return self.additional_properties[key]

    def __setitem__(self, key: str, value: Any) -> None:
        self.additional_properties[key] = value

    def __delitem__(self, key: str) -> None:
        del self.additional_properties[key]

    def __contains__(self, key: str) -> bool:
        return key in self.additional_properties

File: dqops/src/main/java/com/dqops/rest/models/metadata/DefaultColumnChecksPatternModel.java
/*
 * Copyright © 2021 DQOps (support@dqops.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.dqops.rest.models.metadata;

import com.dqops.checks.column.checkspecs.nulls.ColumnNullsCountCheckSpec;
import com.dqops.checks.column.monitoring.ColumnDailyMonitoringCheckCategoriesSpec;
import com.dqops.checks.column.monitoring.ColumnMonitoringCheckCategoriesSpec;
import com.dqops.checks.column.monitoring.nulls.ColumnNullsDailyMonitoringChecksSpec;
import com.dqops.metadata.defaultchecks.column.ColumnDefaultChecksPatternSpec;
import com.dqops.metadata.defaultchecks.column.TargetColumnPatternSpec;
import com.dqops.rules.comparison.MaxCountRule0ErrorParametersSpec;
import com.dqops.utils.docs.generators.SampleValueFactory;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import io.swagger.annotations.ApiModel;
import lombok.Data;

/**
 * Default column-level checks pattern model that is returned by the REST API. Describes a configuration of data quality checks for a named pattern. DQOps applies these checks on columns that match the filter.
 */
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@ApiModel(value = "DefaultColumnChecksPatternModel", description = "Default column-level checks pattern model")
public class DefaultColumnChecksPatternModel {
    @JsonPropertyDescription("Pattern name")
    private String patternName;

    @JsonPropertyDescription("The default checks specification.")
    private ColumnDefaultChecksPatternSpec patternSpec;

    /**
     * Boolean flag that decides if the current user can update or delete this object.
     */
    @JsonPropertyDescription("Boolean flag that decides if the current user can update or delete this object.")
    private boolean canEdit;

    /**
     * Optional parsing error that was captured when parsing the YAML file.
     * This field is null when the YAML file is valid. If an error was captured, this field returns the file parsing error message and the file location.
     */
    @JsonPropertyDescription("Optional parsing error that was captured when parsing the YAML file. " +
            "This field is null when the YAML file is valid. If an error was captured, this field returns the file parsing error message and the file location.")
    private String yamlParsingError;

    /**
     * Default constructor for TableDefaultChecksPatternEditModel.
     */
    public DefaultColumnChecksPatternModel() {
    }


    public static class ColumnDefaultChecksPatternEditModelSampleFactory implements SampleValueFactory<DefaultColumnChecksPatternModel> {
        @Override
        public DefaultColumnChecksPatternModel createSample() {
            return new DefaultColumnChecksPatternModel() {{
                setPatternName("id columns not null");
                setPatternSpec(new ColumnDefaultChecksPatternSpec() {{
                    setTarget(new TargetColumnPatternSpec() {{
                        setColumn("id");
                    }});
                    setMonitoringChecks(new ColumnMonitoringCheckCategoriesSpec() {{
                        setDaily(new ColumnDailyMonitoringCheckCategoriesSpec() {{
                            setNulls(new ColumnNullsDailyMonitoringChecksSpec() {{
                                setDailyNullsCount(new ColumnNullsCountCheckSpec() {{
                                    setError(new MaxCountRule0ErrorParametersSpec());
                                }});
                            }});
                        }});
                    }});
                }});
                setCanEdit(true);
            }};
        }
    }
}
